addons/shortcodes.php

```diff
@@ -59,179 +77,280 @@ function shutdown() {
-		), $attr, 'camptix_attendees' );
-
-		$camptix_options = $camptix->get_options();
+	public function cache_all_attendees_shortcodes() {
+        // Get posts containing the `camptix_attendees` shortcode
+		$params = array(
+			'post_type'              => 'page',
+			'post_status'            => 'publish',
+			's'                      => '[camptix_attendees',
+			'posts_per_page'         => 50,
+			'update_post_term_cache' => false,
+			'update_post_meta_cache' => false,
+		);
+		$posts  = get_posts( $params );
 
-		// Enqueue Jetpack's spinner script if available
-		if ( wp_script_is( 'jquery.spin', 'registered' ) ) {
-			wp_enqueue_script( 'jquery.spin' );
+		if ( ! $posts ) {
+			return;
 		}
 
-		// For wp.template()
-		wp_enqueue_script( 'wp-util' );
+		$regex = get_shortcode_regex( array( 'camptix_attendees' ) );
 
-		// Lazy load the camptix js.
+		foreach ( $posts as $post ) {
+			$matches = array();
+
+			if ( ! preg_match_all( $regex, $post->post_content, $matches, PREG_SET_ORDER ) ) {
+				continue;
+			}
+
+			foreach ( $matches as $match ) {
+				$attr = shortcode_parse_atts( $match[3] );
+				$attr = $this->sanitize_attendees_atts( $attr );
+
+                $this->get_attendees_shortcode_content( $attr );
+			}
+		}
+	}
+
+	/**
+	 * Callback for the [camptix_attendees] shortcode.
+	 */
+	public function shortcode_attendees( $attr ) {
+		// Required scripts
+		wp_enqueue_script( 'wp-util' ); // For wp.template()
+		if ( wp_script_is( 'jquery.spin', 'registered' ) ) {
+			wp_enqueue_script( 'jquery.spin' ); // Enqueue Jetpack's spinner script if available
+		}
 		wp_enqueue_script( 'camptix' );
 
 		// Only print the JS template once.
 		if ( ! has_action( 'wp_print_footer_scripts', array( $this, 'avatar_js_template' ) ) ) {
 			add_action( 'wp_print_footer_scripts', array( $this, 'avatar_js_template' ) );
 		}
 
-		$start = microtime(true);
+		$attr = $this->sanitize_attendees_atts( $attr );
+
+		/**
+		 *
+		 */
+		do_action( 'camptix_attendees_shortcode_init', $attr );
+
+		return $this->get_attendees_shortcode_content( $attr );
+	}
+
+	/**
+	 * Generate the key for a particular configuration to use when
+	 * setting or retrieving a cached value.
+	 *
+	 * @param array $attr    Sanitized shortcode attributes
+	 *
+	 * @return string        The cache key
+	 */
+	protected function generate_attendees_cache_key( $attr ) {
+		return 'camptix-attendees-' . md5( maybe_serialize( $attr ) );
+	}
+
+	/**
+	 * Get the content for an instance of the [camptix_attendees] shortcode.
+	 *
+	 * This checks for a cached version first. If none is found, it generates
+	 * the content and caches it before returning.
+	 *
+	 * @param array $attr             Sanitized shortcode attributes
+	 * @param bool  $force_refresh    True to generate the content even if cached value is found.
+	 *
+	 * @return string                 Rendered shortcode content
+	 */
+	public function get_attendees_shortcode_content( $attr, $force_refresh = false ) {
```

Comment:
It doesn't look like `$force_refresh` is ever used. Is there a reason to keep it in?